Fix the OGM page (#431)
neo4j/docs-getting-started: modules/ROOT/pages/languages-guides/java/neo4j-ogm.adoc

[#ogm-faq]
== FAQ

What is the difference between Neo4j-OGM and Spring Data Neo4j (SDN)?::

> Spring Data Neo4j (SDN) up to version 5.3.x uses Neo4j-OGM under the covers. It’s like Spring Data JPA, where JPA/Hibernate is the underlay. Most of the power of SDN actually comes from Neo4j-OGM. Starting with Spring Data Neo4j 6.x (Spring Boot 2.4) does not need Neo4j-OGM anymore. Please note that, intentionally, Neo4j-OGM 4+ does not work as a drop-in replacement for Spring Data 5.x.

Reviewer: ```suggestion
Spring Data Neo4j (SDN) up to version 5.3.x uses Neo4j-OGM under the covers. 
It’s like Spring Data JPA, where JPA/Hibernate is the underlay.

Most of the power of SDN actually comes from Neo4j-OGM.
However, starting with Spring Data Neo4j 6.x (Spring Boot 2.4), Neo4j-OGM is not needed anymore. 

Please note that, intentionally, Neo4j-OGM 4+ does not work as a drop-in replacement for Spring Data 5.x.
```

> 

Reviewer: For some reason there is too little space between the two answers. Maybe this should help.

```suggestion

```

How are labels generated when using inheritance?::

> All concrete classes generate a label, abstract classes and interfaces not. If any kind of class or interface gets annotated with @NodeEntity or @NodeEntity(label="customLabel") it will generate a label. Any class annotated with @Transient will not generate a label.

Reviewer: ```suggestion
All concrete classes generate a label, but abstract classes and interfaces not. 
If any kind of class or interface gets annotated with `@NodeEntity` or `@NodeEntity(label="customLabel")`, it will generate a label. 
Any class annotated with `@Transient` will not generate a label.
```

| icon:book[] Docs | https://neo4j.com/docs/ogm-manual/current/
|===

== Compatibility

> Consult the version table to determine which version of Neo4j-OGM to use with a particular version of Neo4j and related technologies.

Reviewer: ```suggestion
Check the table to determine which version of Neo4j-OGM to use with a particular version of Neo4j and related technologies:
```

[frame="topbot",options="header"]
|======================
|Neo4j-OGM Version   |Neo4j Version^1^
|4.0.x^2^            |4.4.x^6^, 5.x
|3.2.x               |3.2.x, 3.3.x, 3.4.x, 3.5.x, 4.0.x^2^, 4.1.x^2^, 4.2.x^2^, 4.3.x^2,5^, 4.4.x^2,5^
|3.1.x^3^            |3.1.x, 3.2.x, 3.3.x, 3.4.x
|3.0.x^3^            |3.1.9, 3.2.12, 3.3.4, 3.4.4
|2.1.x^4^            |2.3.9, 3.0.11, 3.1.6
|2.0.2^4^            |2.3.8, 3.0.7
|2.0.1^4^            |2.2.x, 2.3.x
|======================

^1^ __The latest supported bugfix versions.__ +

> ^2^ __These versions only support connections via Bolt.__ +

Reviewer: ```suggestion
^2^ __Only supports connections via Bolt.__ +
```

> ^3^ __These versions are no longer actively developed.__ +

Reviewer: ```suggestion
^3^ __No longer actively developed.__ +
```

> ^4^ __These versions are no longer actively developed or supported.__ +

Reviewer: ```suggestion
^4^ __No longer actively developed or supported.__ +
```

^5^ __Neo4j-OGM 3.2.24+ only.__ +

> ^6^ __Technical working but not officially supported__ +

Reviewer: ```suggestion
^6^ __Technically working, but not officially supported__ +
```
